Add config to truncate commit hashes when copying them to the clipboard

stefanhaller opened the pull request:
- **PR Description**

I often copy hashes in the commits panel in order to paste them into Github comments (or other places), and I can't stand it when they have the full length.

I picked a default of 12 for this; I find this to be a good middle ground between being reliable in large repos (12 still works in the linux kernel repo today, but it might not be enough in really huge repos) and not being too ugly (many smaller repos can probably get away with less).

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

karimkhaleel commented:
Code makes sense. You think we should do the same thing for the `y-enter` combo?

![image](https://github.com/jesseduffield/lazygit/assets/94621779/24de1bdd-1892-4f5e-8c07-34347553a258)

stefanhaller commented:
> You think we should do the same thing for the `y-enter` combo?

Ah good point, I hadn't thought of that as I never use it. Yes, probably makes sense, see 3745ee1ac0ab.

Wait, on the other hand, dropping that would give people a way to get the unabbreviated one if they need it only occasionally. Dunno. We could even have two entries in the `y` menu, "Full hash" and "Abbreviated hash", but maybe that's taking it too far.

I'm undecided, somebody please tell me what to do. 😄

karimkhaleel commented:
I honestly kind of like the idea of just dropping that fixup. That entry is a convenient way to get the full hash for people that might need it.

But we should at least change the log message that gets shown from "Copy commit SHA/hash to clipboard" to something like "Copy full commit SHA/hash to clipboard".

stefanhaller commented:
> I honestly kind of like the idea of just dropping that fixup. That entry is a convenient way to get the full hash for people that might need it.

Agreed, dropping fixup again.

> But we should at least change the log message that gets shown from "Copy commit SHA/hash to clipboard" to something like "Copy full commit SHA/hash to clipboard".

Done in 19d6dc68bb84. (Although I don't find this super important, since the line right below shows that the full hash was copied. But anyway.)

I find it more important to change the toast in the status bar to show what was copied, for those who hide the log window (I do). Changed in f65ea42c92f8.

Also added a minor fix to the casing of that toast, see 222a9a871727.

If you wanna have another look at these, @karimkhaleel?

karimkhaleel commented:
Perfect :+1:
